ocs_ci/ocs/pgsql.py

```diff
@@ -663,3 +664,193 @@ def pgsql_full(self):
         pgbench_pods = self.get_pgbench_pods()
         # Validate pgbench run and parse logs
         self.validate_pgbench_run(pgbench_pods)
+
+    def generate_random_date(self, min_year=1900, max_year=datetime.now().year):
+        """
+        Generate a random date in yyyy-mm-dd format.
+
+        Args:
+            min_year (int) : The minimum year for which the date can be generated.
+            max_year (int) : The maximum year for which the date can be generated.
+
+        Returns:
+            str : Random date in yyyy-mm-dd format.
+
+        """
+        start = datetime(min_year, 1, 1, 00, 00, 00)
+        years = max_year - min_year + 1
+        end = start + timedelta(days=365 * years)
+        return (start + (end - start) * random.random()).strftime("%Y-%m-%d")
+
+    def generate_random_string(self, length):
+        """
+        Generate a random string of given length.
+
+        Args:
+            length (int) : The length of the string that needs to be generated.
+
+        Returns:
+            str : Random string of given length.
+
+        """
+        return "".join(random.choices(string.ascii_uppercase + string.digits, k=length))
+
+    def run_pgsql_command(self, postgres_pod, command, select=False):
+        """
+        Run pgsql command on the postgres POD
+
+        Args:
+            postgres_pod (POD) : POD object on which postgres is running.
+            command (str) : The pgsql command that we want to run.
+            select (bool) : True if the command is of type select, False otherwise.
+
+        Returns:
+            str : Output of the select query.
+
+        """
+        out = postgres_pod.exec_cmd_on_pod(f'psql -U postgres -c "{command}" ')
+        if select:
+            ind = out.index("-")
+            result = list()
+            result.append("\n" + out[:ind])
+            result.append("-" * len(result[0]))
+            result.extend(
+                re.findall(r"\d+\s\|\s[A-Z0-9]+\s\|\s\d{4}-\d{2}-\d{2}\s", out)
+            )
+            result.append("-" * len(result[0]))
+            log.info("\n".join(result))
+            return out
```

Comment:
It is better to return the command output in any case and not for select only.

Reply:
Done